### DISCUSSION TITLE
Audiobooks.bundle obsolete?

### DISCUSSION THREAD

**Discussion by rabelux** [seanap/Plex-Audiobook-Guide#29] [General]
@seanap reading your last comment on [#5](https://github.com/seanap/Audiobooks.bundle/issues/5) I get the impression there is no real benefit from using the bundle anymore? Considering plugins are being phased out ([1]) I'd be glad if that's the case.

I mean you have an awesome setup for mp3tag that seems to fill all the necessary fields.
What does the plugin add to it?

[1]:https://forums.plex.tv/t/i-cant-add-update-plugins-agents-metadatas-with-dsm7-help/732576/11

**Comment by seanap** [comment]
Plug-in's are already phased out. Plex wanted to stop users from using plug-ins to support the blatantly illegal web streaming sites.  That said you can still use the plug-ins for utilities such as metadata agents you just have to drop the files into the appdata folder manually.  There are no plans to remove this type of use anytime soon.  Plex is not getting rid of the ability to run these types of plugins, but by "Unsupported" they will no longer help if you have issues and plugins break things.  The post linked is specifically for the newest update for the Synology Plex Server app, and I would suspect that Synology broke something with the update. Many people continue to use the audiobook metadata agent on Linux and Windows.  The fix for synology is out of my hands but it sounds like a permissions issue (and an access issue to access the appdata folders) and not depreciated functionality.  

The Audiobooks.bundle is not obsolete, it serves an important function and should be used in conjunction with properly tagged files to get the most amount of metadata into plex.  Plex does not read all of the id3 tags. The only way to get certain data into plex is through an Agent.  The full list of tags that the agent sets can be found in the readme of my Audiobook Agent repo.

https://github.com/seanap/Audiobooks.bundle#metadata-supplied-to-plex

**Reply by seanap** [reply]
That said, I don't have any insider information, and Plex can add or remove features at anytime.  I just have never heard of any attempt to remove the ability for custom scanners and agents.

**Reply by seanap** [reply]
Maybe you linked the wrong support issue but #10 was specifically about the web scraper.  Since we don't have a database we can pull from we have to rely on web scraping which can easily break if Amazon updates their underlying web page source code.  They have done this many times in the past, and will do it many times in the future, but its a relatively easy fix to remap the webscraper. This will effect the metadata agent when this happens as well.

**Reply by rabelux** [reply]
Thanks for your fast and thorough reply! I indeed linked the wrong issue and fixed it in the original post.

**Reply by seanap** [reply]
Ah, Yeah, that was about the new Synology update.  It sounds like the update moved the location of the appdata folder, and there was an update which brought it back, but when you have to access it as root you need to make very sure that you know the chmod and chown of those folders and when you add the Audiobooks.bundle folder it and all the files have to match.  Very easy to get the wrong permissions.

Unfortunately I don't have a Synology to test. It looks like there have been people to get the scanner and agents to work on DSM 7 by moving the folder manually through SSH and setting the permissions. DSM 7 mandated a storage structure change which removed the “Plex” shared folder, this was not an intentional removal of plug-ins just a storage restructuring that happened to break plug-ins.

https://forums.plex.tv/t/solved-dsmv7-how-to-update-plugins-agents-scanners/700715/12

**Comment by hoelkmen** [comment]
Hi,

im using PLex in an Ubuntu Docker Container  (Version 1.24.3.5033). 

I Downloaded the Bundle and put it in de "Plug-Ins" Folder. But the Agent is nowhere to be found after reboot.

Not in The "Agents Settings" and not in the Library Creation Menu.

I Dont know what to do, or what i did wrong. 

Do you have any idea / guidance ?

Thank you!

**Reply by rabelux** [reply]
Two reasons I can think of:
- your folder name doesn't end with `.bundle`
- your container uses a different application directory

**Reply by hoelkmen** [reply]
Oh, man. Thanks for the quick reply.

My mistake. I had an old  mount point in my plex-docker. After i fixed this i can finally use the Audiobook agent.

Thanks a lot.

**Comment by seanap** [comment]
Make sure the folder structure is like this:

`../Plug-ins/Audiobooks.bundle/[CODE FILES]`

Sometimes you can have nested Audiobooks.bundle folders, or like rabelux said you have to delete the "-master" in the foldername.

Also, make sure you update the PMS version to 1.24.4, the version you listed has a major bug that breaks the agent (but shouldn't have prevented it from showing up)

Permissions could also be an issue, you can try a `chmod -R 755 Audiobooks.bundle` and a `chown -R plex:plex Audiobooks.bundle`

If using MergerFS there might be a CoW (Copy-on-Write) issue.

**Reply by hoelkmen** [reply]
Thank you too for your support. As i mentioned above,  i had the wrong mount point in my container.

I am using the linuxserver.io docker container for my pms. I am going to update this as soon as they push a new container verision.

greetings

**Reply by hoelkmen** [reply]
But what do you mean with major bug?

Like this: "Album Name" by Author.. this looks weird...

![image](https://user-images.githubusercontent.com/63524270/135665701-d57f2d95-9f20-48f4-9169-5528fa27b472.png)

**Reply by seanap** [reply]
Yes, update to latest PMS. The `linuxserver/plex:latest` has the fix
![image](https://user-images.githubusercontent.com/17012946/135725157-a63b3903-5584-4b35-a126-0c44320b7702.png)

**Reply by hoelkmen** [reply]
Yes, no ist works fine!

Thänk you again
